components/dashboard/UpcomingInterviewDialog/formValidationHook.js

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from 'react';
+
+const useFormValidation = (initialState, validate, runOnSubmit) => {
+  const [values, setValues] = useState(initialState);
+  const [errors, setErrors] = useState({});
+
+  const [touched, setTouched] = useState([]);
+  const [isSubmitting, setSubmitting] = useState(false);
+
+  useEffect(() => {
+    if (isSubmitting) {
+      const noErrors = Object.keys(errors).length === 0;
+      if (noErrors) {
+        setTouched([]);
+        runOnSubmit(values);
+        setSubmitting(false);
+      } else {
+        setSubmitting(false);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Similar to below -- let's find a way that observes `exhaustive-deps`.

Again here, it feels a little roundabout to perform the actual submit in an effect that looks for a flag. Couldn't all be synchronous in the submit click handler?

Reply:
Happy to jam on this with you, if you'd like (but feel free to run with it without me).

Reply:
I see what you are saying, but we definitely want the isSubmitting to be a part of the hook's state. We also want to make sure the effect is run whenever the errors object is updated -- the check for isSubmitting is ensuring that the submit is not run when the component is first rendered and the errors object is empty.